docs/builder_image.md

```diff
@@ -116,9 +115,7 @@ The `run` script is responsible for executing your application.
 
 ## save-artifacts
 
-The `save-artifacts` script is responsible for gathering all the dependencies which
-existence can speed up the following build processes (eg. for Ruby - gems installed by Bundler,
-for Java - `.m2` contents, etc.) into a tar file and stream it to the standard output.
+The `save-artifacts` script is responsible for gathering all the dependencies into a tar file and streaming it to the standard output.  The existance of this can speed up the following build processes (eg. for Ruby - gems installed by Bundler, for Java - `.m2` contents, etc.).
```

Comment:
I'd suggest moving the text from parenthesis after the first sentence, as it's talking about the contents of the tar, like following:

 The `save-artifacts` script is responsible for gathering all the dependencies into a tar file and streaming it to the standard output (eg. for Ruby - gems installed by Bundler, for Java - `.m2` contents, etc.).  The existance of this can speed up the following build processes.